app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,332 +1,367 @@
+                }
 
-            updateTextColors(mainHolder)
-            if (stroboscopeBar.isInvisible()) {
-                changeIconColor(contrastColor, stroboscopeBtn)
+                finalItems.sortBy { it.id }
+                finalItems.add(RadioItem(-1, getString(com.simplemobiletools.commons.R.string.custom)))
+                finalItems.toImmutableList()
             }
         }
 
-        binding.sleepTimerHolder.background = ColorDrawable(getProperBackgroundColor())
-        binding.sleepTimerStop.applyColorFilter(getProperTextColor())
-
-        requestedOrientation = if (config.forcePortraitMode) ActivityInfo.SCREEN_ORIENTATION_PORTRAIT else ActivityInfo.SCREEN_ORIENTATION_SENSOR
-        invalidateOptionsMenu()
+        RadioGroupAlertDialog(
+            alertDialogState = alertDialogState,
+            items = items,
+            selectedItemId = preferences.lastSleepTimerSeconds,
+            callback = {
+                if (it as Int == -1) {
+                    onCustomValueSelected()
+                } else if (it > 0) {
+                    pickedSleepTimer(it)
+                }
+            }
+        )
+    }
 
-        if (config.turnFlashlightOn && reTurnFlashlightOn) {
-            mCameraImpl!!.enableFlashlight()
+    @Composable
+    private fun AppLaunched(
+        donateAlertDialogState: AlertDialogState = getDonateAlertDialogState(),
+        rateStarsAlertDialogState: AlertDialogState = getRateStarsAlertDialogState(),
+    ) {
+        LaunchedEffect(Unit) {
+            appLaunchedCompose(
+                appId = BuildConfig.APPLICATION_ID,
+                showDonateDialog = donateAlertDialogState::show,
+                showRateUsDialog = rateStarsAlertDialogState::show,
+                showUpgradeDialog = {}
+            )
         }
-
-        reTurnFlashlightOn = true
-
-        checkShortcuts()
     }
 
-    override fun onStart() {
-        super.onStart()
-        mBus!!.register(this)
-
-        if (config.sleepInTS == 0L) {
-            binding.sleepTimerHolder.beGone()
-            (getSystemService(Context.ALARM_SERVICE) as AlarmManager).cancel(getShutDownPendingIntent())
+    @Composable
+    private fun getDonateAlertDialogState() =
+        rememberAlertDialogState().apply {
+            DialogMember {
+                DonateAlertDialog(alertDialogState = this)
+            }
         }
 
-        if (mCameraImpl == null) {
-            setupCameraImpl()
+    @Composable
+    private fun getRateStarsAlertDialogState() = rememberAlertDialogState().apply {
+        DialogMember {
+            RateStarsAlertDialog(alertDialogState = this, onRating = ::rateStarsRedirectAndThankYou)
         }
     }
 
-    override fun onStop() {
-        super.onStop()
-        mBus!!.unregister(this)
-    }
-
-    override fun onDestroy() {
-        super.onDestroy()
-        releaseCamera()
+    @Composable
+    private fun getCameraPermissionLauncher(
```

Comment:
we don't need this function at all, it's just a provision function with a really bad name